pkg/prom/instance/instance.go

```diff
@@ -253,7 +255,7 @@ func newInstance(globalCfg config.GlobalConfig, cfg Config, reg prometheus.Regis
 // Run may be re-called after exiting, as components will be reinitialized each
 // time Run is called.
 func (i *Instance) Run(ctx context.Context) error {
-	level.Debug(i.logger).Log("msg", "running instance", "name", i.cfg.Name)
+	level.Debug(i.logger).Log("msg", "initializing instance", "name", i.Config().Name)
```

Comment:
nit: I'd just use `i.cfg` directly if these fields are not allowed to change whilst updating then there's really no need to take the lock. The same comment applies to other uses around with similar constrains.